flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/OperatorSubtaskState.java

```diff
@@ -59,6 +60,8 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(OperatorSubtaskState.class);
 
+    public static final OperatorSubtaskState FINISHED = new OperatorSubtaskState(true);
+
     private static final long serialVersionUID = -2394696997971923995L;
```

Comment:
Yes, since users may upgrade from the older version, we might need to keep the compatibility. 

But to avoid unexpected problems I change the implementation to not modify this class, but introduce a new `FinishedOperatorSubtaskState` subclass specially.